native-addons-are-close-to-being-on-par-with-js-modules.md

```diff
@@ -0,0 +1,113 @@
+
+Although originally this change was made with the intention of allowing well-known symbols for multiple versions of Node.js to co-habitate within the same binary, it also enables multiple loading. Nevertheless, the ability to load a native addon multiple times means that the addon must adhere to a [context-aware addon][] structure.
+
+### 5. Thread Safety
+Since JavaScript modules are self-contained, even given their own scope at module load time by Node.js, they raise no thread safety concerns. In contrast, native addons must avoid global state in order to be considered thread-safe. This requires an explicit design decision on the part of the maintainer. Fortunately, per-addon-instance data can be heap-allocated at module initialization time, and passed to each native binding made available from the addon as seen in the example of the [context-aware addon][]. This applies to both N-API addons and V8 addons.
+
+
+## Conclusion
+Although the need for rendering a native addon as multiple binaries, one for each platform and architecture it supports, cannot be eliminated, with recent Node.js improvements, we can consider native addons as having taken a significant step towards being "on-par" with JavaScript modules:
+
+|  |  | JavaScript modules | Native addons then | Native addons now |
+|--- |--- |--- |--- |--- |
+| 1. | ... need to be compiled. | No | Not if there are pre-builds | Not if there are pre-builds |
+| 2. | ... will work on all platforms and architectures. | Yes | Yes if there are pre-builds | Yes if there are pre-builds |
+| 3. | ... will work with all current and subsequent Node.js versions once written. | Yes | No | Yes if using N-API |
+| 4. | ... can be loaded multiple times. | Yes | No | Yes if written as a [context-aware addon][] |
+| 5. | ... are thread-safe if not explicitly making use of threading infrastructure. | Yes | No | Yes if written as a [context-aware addon][] |
+
+The picture that emerges for native addon maintainers is that they now have the tools to provide application maintainers with native addon packages that
+* need not be re-published by addon maintainers whenever a new version of Node.js is released, and that
+* need not be re-installed by application maintainers solely because they wish to drop a new version of Node.js into their production invironment, and that
+* can be used safely in a multi-threaded JavaScript execution environment.
+
+It takes an effort on the part of native addon maintainers to make use of these new tools, however, the payoff is that as a result of such an effort they will not have to re-publish their packages in response to a new Node.js release. For application maintainers the availability of packages that take advantage of these new tools means that, as soon as all their dependencies are satisfied by native addons that are enabled using these tools, they too can avoid a rebuild/redeploy in response to new Node.js releases.
```

Comment:
"For application maintainers the" should better be "For application maintainers, the", i.e. a comma would ease the read, imho..